### DISCUSSION TITLE
About polymerization process simulation on DWSIM

### DISCUSSION THREAD

**Discussion by Jackxi1024** [DanWBR/dwsim#202] [General]
Hello, I'm doing research on polymerization process simulation recently. I found that DWSIM is a great project. I want to complete my research based on DWSIM.
Therefore, I want to define polymers (such as polyethylene) and corresponding properties (such as number average molecular weight), which is not available on DWSIM. I need the PC-SAFT equation of state as the thermodynamic model, which is available on DWSIM.
In addition, I want to rewrite PFR reactor and CSTR reactor with Python script to calculate polymerization kinetics. I don't know if the python script in DWSIM can establish and solve differential equations.
I would appreciate it if you could provide some suggestions.

**Comment by kookma** [comment]
@Jackxi1024 - I think the best and may be the fastest method to implement polymerization reactions and reactor in DWSIM is to use Python + DWSIM. A good start point is using the python script module to develop a custom unit operation!
You can access to stream data, thermodynamics properties,... from Script Manager inside DWSIM. 
You have have sophisticated computational modules in python and just call them from inside DWSIM script manager! 
You can use and  high quality python libraries like Numpy, Scipy, Matplotlib, ...

I may recommend to have a look here to master how Python + DWSIM works

- https://dwsim.fossee.in/custom-model/completed-custom-models (a wealth of example of using python custom uo)
- https://sourceforge.net/p/dwsim/discussion/scripting/ (some Q&A on using python in dwsim)
- https://dwsim.inforside.com.br/wiki/index.php?title=Tutorials
- https://dwsim.inforside.com.br/tw/import-user-modules/import-user-module.html

**Comment by Jackxi1024** [comment]
Thank you very much for your reply.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "DanWBR/dwsim6"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年9月16日(星期四) 中午12:48
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [DanWBR/dwsim6] About polymerization process simulation on DWSIM (Discussion #202)





 
@Jackxi1024 - I think the best and may be the fastest method to implement polymerization reactions and reactor in DWSIM is to use Python + DWSIM. A good start point is using the python script module to develop a custom unit operation!
 You can access to stream data, thermodynamics properties,... from Script Manager inside DWSIM.
 You have have sophisticated computational modules in python and just call them from inside DWSIM script manager!
 You can use and  high quality python libraries like Numpy, Scipy, Matplotlib, ...
 
I may recommend to have a look here to master how Python + DWSIM works
  
https://dwsim.fossee.in/custom-model/completed-custom-models (a wealth of example of using python custom uo)
 
https://sourceforge.net/p/dwsim/discussion/scripting/ (some Q&amp;A on using python in dwsim)
 
https://dwsim.inforside.com.br/wiki/index.php?title=Tutorials
 
https://dwsim.inforside.com.br/tw/import-user-modules/import-user-module.html
  
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
